pgmpy/tests/test_models/test_FactorGraph.py

```diff
@@ -121,6 +123,22 @@ def test_get_uniform_message(self):
         message = self.graph.get_uniform_message("a")
         assert (message == np.array([0.25, 0.25, 0.25, 0.25])).all()
 
+    def test_add_edge_sets_default_weight(self):
+        G = FactorGraph()
+        G.add_nodes_from(["a", "b"])
+
+        phi1 = DiscreteFactor(["a", "b"], [2, 2], np.ones(4))
+        G.add_node(phi1)
+
+        G.add_edge("a", phi1)
+        self.assertEqual(G.get_edge_data("a", phi1)["weight"], 0)
+
+        try:
```

Comment:
I don't think we need to plot the graph, as we can't check that it is drawn correctly.